realm/src/main/java/io/realm/Realm.java

```diff
@@ -657,6 +658,14 @@ private static void validateAgainstExistingConfigurations(RealmConfiguration new
                 throw new IllegalArgumentException("Two configurations with different schemas are trying to open " +
                         "the same Realm file. Their schema must be the same: " + newConfiguration.getPath());
             }
+
+            // Check if the durability is the same
+            SharedGroup.Durability cachedDurability = cachedConfiguration.getDurability();
+            SharedGroup.Durability newDurability = newConfiguration.getDurability();
+            if (!cachedDurability.equals(newDurability)) {
+                throw new IllegalArgumentException("A Realm cannot be both in-memory and persisted. Two conflicting " +
+                        "configurations pointing to " + newConfiguration.getRealmFileName() + " are being used.");
```

Comment:
Hmm. My initial reaction would probably be that you would rarely mix inMemory and persisted Ream anyway, and even if you did it accidentally we have the configuration check in Realm.java that will throw an error in that case. So no, I don't think we need to explicitly require a name for in-memory Realms.